### PR TITLE
Task4

### PR DESCRIPTION
**Pull request by GGnk**
## Evaluation criteria

1. Implementation of markup and styles; 
2. Use stateless/stateful approach; 
3. Use React synthetic events;
4. Use lifecycle methods (discuss with mentor). 

### Demo - [web page](https://react-nurlan.netlify.app)

### CODE REVIEW

File: src/components/Header.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    genres: [],
+    releaseDate: ''
+}
+
+const Header: React.FC = () => {
+    const [showModal, setShowModal] = useState(false);
+
+    const modal = showModal 
+                        && (
+                        <Suspense fallback={<div>Загрузка...</div>}>
```
Comment (mmarkelov) [inline src/components/Header.tsx]: If you are use english, it's better to rename to `Loading`

File: src/components/Main.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    'release date', 'best rating'
+]
+
+const Main: React.FC = () => {
+    const [select, setSelect] = useState('release date')
+    
+    const listCategory = category.map((cat: string) => <li key={cat}>{cat}</li>)
+
+    const listFilters = filters.map((filter: string, i) => {
+        return <option key={i} value={filter}>{filter.toUpperCase()}</option>
```
Comment (mmarkelov) [inline src/components/Main.tsx]: Maybe it's better to use `filter` for `key`?

File: src/components/blocks/Form.tsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+type Form = {
+    film: Movie,
+    setShowModal: (arg: boolean) => void
+}
+
+const Form: React.FC<Form> = ({ film, setShowModal }) => {
+    const [movie, setMovie] = useState<Movie>({...film})
+
+    const handleGenres = (event: { target: { options: HTMLOptionsCollection } }) => {
+        let options = event.target.options;
```
Comment (mmarkelov) [inline src/components/blocks/Form.tsx]: Maybe you can use smth like this?
```js
const selectedOptions = options.filter(option => option.selected).map(item => item.value)
```

File: src/components/blocks/Modal.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+import { createPortal } from 'react-dom';
+
+const Modal = (props: any) => {
+    const el = document.createElement('div');
+    const modal = document.getElementById('modal') as HTMLHtmlElement;
+
+    useEffect(():any => {
+        modal.appendChild(el);
+        return () => modal.removeChild(el);
+    });
```
Comment (mmarkelov) [inline src/components/blocks/Modal.tsx]: Did you miss empty array here as dependencies?

File: src/components/Header.tsx (hunk 1 above)
Comment (GGnk) [reply src/components/Header.tsx]: fixed it

File: src/components/Main.tsx (hunk 2 above)
Comment (GGnk) [reply src/components/Main.tsx]: fixed it

File: src/components/blocks/Form.tsx (hunk 3 above)
Comment (GGnk) [reply src/components/blocks/Form.tsx]: fixed it

File: src/components/blocks/Modal.tsx (hunk 4 above)
Comment (GGnk) [reply src/components/blocks/Modal.tsx]: fixed it

**Pull request closed without merging**